packages/vite/src/node/ssr/runtime/runtime.ts

```diff
@@ -218,21 +218,20 @@ export class ViteRuntime {
         return this.processImport(mod.exports, fetchedModule, metadata)
     }
 
-    const getStack = () =>
-      `stack:\n${[...callstack, moduleId]
-        .reverse()
-        .map((p) => `  - ${p}`)
-        .join('\n')}`
-
     let debugTimer: any
-    if (this.debug)
-      debugTimer = setTimeout(
-        () =>
-          this.debug!(
-            `[vite-runtime] module ${moduleId} takes over 2s to load.\n${getStack()}`,
-          ),
-        2000,
-      )
+    if (this.debug) {
+      debugTimer = setTimeout(() => {
+        const getStack = () =>
+          `stack:\n${[...callstack, moduleId]
+            .reverse()
+            .map((p) => `  - ${p}`)
+            .join('\n')}`
```

Comment:
Moved this `getStack` function inside to avoid creating this function when `this.debug` is `false`.